### PR TITLE
Add blogpost on JAAS realm

### PR DESCRIPTION
**Pull request by Skyllarr**

### CODE REVIEW

File: _posts/2021-12-16-jaas-realm.adoc [hunk 1]
```diff
@@ -0,0 +1,86 @@
+---
+layout: post
+title: 'Using JAAS realm in Elytron'
+date: 2021-12-16
+tags: elytron wildfly JAAS security realm
```
Comment (fjuma) [inline _posts/2021-12-16-jaas-realm.adoc:5]: With Ashpan's updates to the tags (see https://github.com/wildfly-security/wildfly-elytron/pull/1634), we can remove the elytron and wildfly tags here and then update the others to "jaas" and "security-realm".

File: _posts/2021-12-16-jaas-realm.adoc [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+synopsis: An example of JAAS realm usage in WildFly.
+author: dvilkola
+---
+
+Elytron subsystem already provides various security realms like LDAP realm, JDBC realm, filesystem realm, and others for common use cases. Starting from WildFly 26, you can also configure JAAS security realm in Elytron subsystem in order to use custom Login Modules for authentication and authorization.
```
Comment (fjuma) [inline _posts/2021-12-16-jaas-realm.adoc:10]: Just a minor comment, s/Elytron subsystem/The Elytron subsystem

Comment (fjuma) [reply _posts/2021-12-16-jaas-realm.adoc:10]: s/JAAS security realm/JAAS security realms

File: _posts/2021-12-16-jaas-realm.adoc [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+JAAS realm uses a https://docs.oracle.com/en/java/javase/11/docs/api/java.base/javax/security/auth/login/LoginContext.html[LoginContext] initialized from JAAS configuration file that will use underlying Login Modules to authenticate and authorize users. Flags and options can be specified in the JAAS configuration file according to the https://docs.oracle.com/javase/8/docs/technotes/guides/security/jgss/tutorials/LoginConfigFile.html[Java documentation].
```
Comment (fjuma) [inline _posts/2021-12-16-jaas-realm.adoc:12]: A JAAS realm

Comment (fjuma) [reply _posts/2021-12-16-jaas-realm.adoc:12]: s/JAAS configuration file/a JAAS configuration file

File: _posts/2021-12-16-jaas-realm.adoc [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+/subsystem=elytron/jaas-realm=jaasRealmExample:add(
+        entry=Entry1,
+        path=/path/to/JAAS-config-file.conf,
+        module=moduleContainingCustomImpls,
+        callback-handler=loginmodules.CustomCallbackHandler)
+```
+
+== Principals to attributes mapping
+
+Elytron subsystem can have attributes associated with authenticated users. These attributes can be for example a name, email, phone number.
```
Comment (fjuma) [inline _posts/2021-12-16-jaas-realm.adoc:39]: The Elytron subsystem

**Comment by fjuma** [comment]
Great post, @Skyllarr! I've just added some small comments.

**Comment by Skyllarr** [comment]
@fjuma I fixed also other ocurrences without a / the, eg the title is now: An example of **a** JAAS realm usage in WildFly.   I hope that is correct. Thank you!

File: _posts/2021-12-16-jaas-realm.adoc (hunk 4 above)
Comment (Skyllarr) [reply _posts/2021-12-16-jaas-realm.adoc]: Fixed.

File: _posts/2021-12-16-jaas-realm.adoc (hunk 3 above)
Comment (Skyllarr) [reply _posts/2021-12-16-jaas-realm.adoc]: Fixed.

File: _posts/2021-12-16-jaas-realm.adoc (hunk 2 above)
Comment (Skyllarr) [reply _posts/2021-12-16-jaas-realm.adoc]: Fixed.

File: _posts/2021-12-16-jaas-realm.adoc (hunk 1 above)
Comment (Skyllarr) [reply _posts/2021-12-16-jaas-realm.adoc]: Fixed.

File: _posts/2021-12-16-jaas-realm.adoc (hunk 3 above)
Comment (Skyllarr) [reply _posts/2021-12-16-jaas-realm.adoc]: Fixed

File: _posts/2021-12-16-jaas-realm.adoc (hunk 1 above)
Comment (Skyllarr) [reply _posts/2021-12-16-jaas-realm.adoc]: changed to login-module and login-context to be consistent, sorry

**Comment by fjuma** [comment]
Perfect, thanks very much @Skyllarr!

For the synopsis, since we are referring to "JAAS realm usage" instead of just "JAAS realm", it should actually be "An example of JAAS realm usage in WildFly" or "An example of how to use a JAAS realm in WildFly".

**Comment by Skyllarr** [comment]
> Perfect, thanks very much @Skyllarr!
> 
> For the synopsis, since we are referring to "JAAS realm usage" instead of just "JAAS realm", it should actually be "An example of JAAS realm usage in WildFly" or "An example of how to use a JAAS realm in WildFly".

@fjuma Fixed

**Comment by Skyllarr** [comment]
I updated the filename and date to be 17th of December.

**Pull request merged**